### PR TITLE
Profile Categories Modal

### PR DESCRIPTION
**Pull request by steam**
Currently `User` does not have any categories to determine the links to display when tapping on the featured category badge button so they are hard coded for now. We'll need to change this as soon as categories are available.

### CODE REVIEW

**Comment by steam** [comment]
@colinta - I updated a bunch of snapshots, we'll need to verify they look correct.

File: Sources/Controllers/Profile/Categories/ProfileCategoriesPresentationController.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+}
+
+// MARK: Presentation
+public extension ProfileCategoriesPresentationController {
+    override public func presentationTransitionWillBegin() {
+        guard let containerView = containerView else { return }
+            background.alpha = 0
+            background.frame = containerView.bounds
+            containerView.addSubview(background)
```
Comment (colinta) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesPresentationController.swift]: Whitespace violation!

File: Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            else {
+                prefix = NSAttributedString(string: " ", attributes: attrs())
+            }
+            featuredIn = featuredIn.append(prefix)
+            featuredIn = featuredIn.append(styleCatgory(category))
+        }
+
+        textView.attributedText = featuredIn
+        textView.sizeToFit()
+        layoutIfNeeded()
```
Comment (steam) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift]: I forgot to remove this. I'll do it now.

File: Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        background.snp_makeConstraints { make in
+            make.edges.equalTo(self)
+        }
+
+        textView.snp_makeConstraints { make in
+            make.leading.trailing.equalTo(self).inset(Size.textInset)
+            make.centerX.centerY.equalTo(self)
+        }
+
+        layoutIfNeeded()
```
Comment (steam) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift]: Here too.

File: Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+    func textViewTapped(link: String, object: ElloAttributedObject) {
+        switch object {
+        case let .AttributedCategory(category):
+            delegate?.categoryTapped(category)
+        default: break
+        }
+    }
+
+    func textViewTappedDefault() {
```
Comment (colinta) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift]: This is the only place I see `textViewTappedDefault` - is it going to be bound to something?

File: Sources/Controllers/Profile/Categories/ProfileCategoriesViewController.swift [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+extension ProfileCategoriesViewController: ProfileCategoriesDelegate {
+
+    public func categoryTapped(category: Category) {
+        print("Tapped \(category.endpoint)")
+
+        let categoryVC = SimpleStreamViewController(endpoint: category.endpoint, title: category.name)
+        categoryVC.currentUser = currentUser
+
+//        presentViewController(categoryVC, animated: true, completion: nil)
```
Comment (steam) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesViewController.swift]: I'll clean this up as well. A few rogue commented out lines.

File: Sources/Controllers/Stream/StreamDataSource.swift [hunk 6]
```diff
@@ -327,9 +327,10 @@ public class StreamDataSource: NSObject, UICollectionViewDataSource {
             (cell as! NotificationCell).webLinkDelegate = webLinkDelegate
             (cell as! NotificationCell).userDelegate = userDelegate
             (cell as! NotificationCell).delegate = notificationDelegate
-        case .ProfileHeader:
-            (cell as! ProfileHeaderCell).simpleStreamDelegate = simpleStreamDelegate
-            (cell as! ProfileHeaderCell).webLinkDelegate = webLinkDelegate
+//        case .ProfileHeader:
+//            (cell as! ProfileHeaderCell).simpleStreamDelegate = simpleStreamDelegate
+//            (cell as! ProfileHeaderCell).webLinkDelegate = webLinkDelegate
```
Comment (steam) [inline Sources/Controllers/Stream/StreamDataSource.swift]: We're probably going to need `webLinkDelegate` added back in.

File: Sources/Controllers/Profile/Categories/ProfileCategoriesProtocols.swift [hunk 7]
```diff
@@ -0,0 +1,11 @@
+////
+///  ProfileCategoriesProtocols.swift
+//
+
+protocol ProfileCategoriesDelegate: class {
+    func categoryTapped(category: Category)
```
Comment (colinta) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesProtocols.swift]: If you added `dismiss` to this protocol, the `UITapGestureRecognizer` could be moved into the `ProfileCategoriesScreen`.

File: Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+        // no-opp
+    }
+}
+
+private extension ProfileCategoriesScreen {
+
+    func styleCatgory(category: Category) -> NSAttributedString {
```
Comment (colinta) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift]: `styleCatgory` -> `styleCategory`

File: Sources/Controllers/Profile/Categories/ProfileCategoriesPresentationController.swift (hunk 1 above)
Comment (steam) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesPresentationController.swift]: 👮

File: Sources/Utilities/InterfaceString.swift [hunk 9]
```diff
@@ -133,7 +133,8 @@ public struct InterfaceString {
         static let CurrentUserNoResultsBody: String = NSLocalizedString("Everything you post lives here!\n\nThis is the place to find everyone you’re following and everyone that’s following you. You’ll find your Loves here too!", comment: "")
         static let NoResultsTitle: String = NSLocalizedString("This person hasn't posted yet.", comment: "")
         static let NoResultsBody: String = NSLocalizedString("Follow or mention them to help them get started!", comment: "")
-        static let TotalViews: String = NSLocalizedString("Total Views", comment: "")
+        static let FeaturedIn: String = NSLocalizedString("Featured in", comment: "Featurd in")
+		static let TotalViews: String = NSLocalizedString("Total Views", comment: "")
```
Comment (colinta) [inline Sources/Utilities/InterfaceString.swift]: Indentation is off here, too. _shakes fist at Xcode_

Comment (colinta) [inline Sources/Utilities/InterfaceString.swift]: Indentation is off here, too. _shakes fist at Xcode_

Comment (colinta) [inline Sources/Utilities/InterfaceString.swift]: Indentation is off here, too. _shakes fist at Xcode_

Comment (colinta) [inline Sources/Utilities/InterfaceString.swift]: Indentation is off here, too. _shakes fist at Xcode_

File: Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift (hunk 4 above)
Comment (steam) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift]: It is part of `ElloTextViewDelegate` and is not optional so I need to have it no-op.

Comment (colinta) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesScreen.swift]: Ahhh!  Thanks.

**Comment by steam** [comment]
Test Suite 'All tests' passed at 2016-10-17 17:00:19.326.
	 Executed 2268 tests, with 0 failures (0 unexpected) in 270.035 (271.704) seconds

File: Sources/Controllers/Profile/Categories/ProfileCategoriesViewController.swift [hunk 10]
```diff
@@ -47,11 +41,13 @@ extension ProfileCategoriesViewController: UIViewControllerTransitioningDelegate
 extension ProfileCategoriesViewController: ProfileCategoriesDelegate {
 
     public func categoryTapped(category: Category) {
-
```
Comment (colinta) [inline Sources/Controllers/Profile/Categories/ProfileCategoriesViewController.swift]: NICE I LIKE THIS LINE THE MOST.

**Pull request merged**